third_party/move/move-compiler-v2/tests/checking/typing/lambda_typed_widen.move

```diff
@@ -0,0 +1,51 @@
+module 0x8675309::M {
+    use std::vector;
+
+    public inline fun use_mut_ref<T>(v: &mut vector<T>, action: |&mut T|T): T {
```

Comment:
Could you write a test case where the return value of the lambda function is immutable/mutable reference?

Reply:
The new tests do reveal some awkward error messages, which can be elicited without this PR, please see #14681.